go/api/config.go

```diff
@@ -401,10 +408,17 @@ func (config *GlideClusterClientConfiguration) WithReadFrom(readFrom ReadFrom) *
 // encompasses sending the request, awaiting for a response from the server, and any required reconnections or retries. If the
 // specified timeout is exceeded for a pending request, it will result in a timeout error. If not set, a default value will be
 // used.
+//
+// WARNING: If requestTimeout exceeds the max duration of 2^32 - 1, a default value will be used.
 func (config *GlideClusterClientConfiguration) WithRequestTimeout(
 	requestTimeout time.Duration,
 ) *GlideClusterClientConfiguration {
-	config.requestTimeout = requestTimeout
+	// Convert back to milliseconds and check if requestTimeout exceeds bounds of a uint32
+	milliseconds := requestTimeout.Milliseconds()
+	if milliseconds < 0 || milliseconds > (1<<32-1) {
+		milliseconds = 5000
+	}
+	config.requestTimeout = uint32(milliseconds)
```

Comment:
The same logic is duplicated multiple times.
Consider refactoring into a function and performing the conversion and bound check where the value is used, e.g. https://github.com/valkey-io/valkey-glide/pull/3933/files#diff-1a739aae3fa68b5778713f3348da86a499f9ee562e59ecfe4d00e4fa0bd01a0fR122.
This way most of the code uses the native duration type making it harder to conflate the units and the conversion is only performed when the value needs to be serialized to the RPC wire format.